homeassistant/components/jewish_calendar/service.py

```diff
@@ -55,24 +56,43 @@ def is_after_sunset(hass: HomeAssistant) -> bool:
 
     async def get_omer_count(call: ServiceCall) -> ServiceResponse:
         """Return the Omer blessing for a given date."""
-        date = call.data.get("date", dt_util.now().date())
+        date = call.data.get(ATTR_DATE, dt_util.now().date())
         after_sunset = (
             call.data[ATTR_AFTER_SUNSET]
-            if "date" in call.data
+            if ATTR_DATE in call.data
             else is_after_sunset(hass)
         )
         hebrew_date = HebrewDate.from_gdate(
             date + datetime.timedelta(days=int(after_sunset))
         )
-        nusach = Nusach[call.data["nusach"].upper()]
         set_language(call.data[CONF_LANGUAGE])
-        omer = Omer(date=hebrew_date, nusach=nusach)
-        return {
-            "message": str(omer.count_str()),
-            "weeks": omer.week,
-            "days": omer.day,
-            "total_days": omer.total_days,
-        }
+
+        try:
+            nusach = Nusach[call.data[ATTR_NUSACH].upper()]
+        except KeyError:
+            raise ServiceValidationError(
+                f"Nusach {call.data[ATTR_NUSACH]} is not supported"
+            ) from None
+
+        try:
+            omer = Omer(date=hebrew_date, nusach=nusach)
+        except ValueError as err:
+            raise HomeAssistantError(
+                f"Unable to calculate the omer for {hebrew_date}"
+            ) from err
+
+        try:
+            return {
+                "message": str(omer.count_str()),
+                "weeks": omer.week,
+                "days": omer.day,
+                "total_days": omer.total_days,
+            }
+        except LookupError as err:
+            raise HomeAssistantError(
```

Comment:
what's a lookup error? When does that happen? It looks like something built in, so can we guard for it?

Reply:
I'm not sure about the whole action exception rule. Since this technically can only happen if the schema failed to validate the Nusach.
Same goes for the other exceptions. I tried reproducing them with tests (due to coverage failing) and every time the schema catches it before I even reach the error.

So maybe I should mark it exempt here? #143763

Reply:
I think we should only catch exceptions that are realistically raised. and looking at this code I am not sure what here would raise this